File: 0283-Move-Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
const moveZeroes = function(nums) {
  let i = 0;
  let length = nums.length;
  
  while(length) {
    if(nums[i] == 0) {
      nums.splice(i,1);
      nums.push(0);
      length--;
      continue;
    }
    
    i++;
    length--;
  }
};

File: solutions/0283-Move-Zeroes.cpp
class Solution {
public:
  void moveZeroes(vector<int>& nums) {
    int s = nums.size();
    auto it = nums.begin();
    
    while(s) {
      if(*it == 0) {
        it = nums.erase(it);
        s--;
        nums.push_back(0);
        continue;
      }
      
      s--;
      it++;
    }
  }
};
